### ISSUE TITLE
Grafana variables doesn't work with victoria metrics

### ISSUE THREAD

**Issue by oOHenry** [jacksontj/promxy#402]
After updating from promxy 0.0.62 to 0.0.68 Grafana Dashboards with variables do not work anymore. With prometheus as datasource it works.
I compared the response of requests against the endpoint: `api/v1/series`. 
Prometheus has the fields `status, data` in the json and victoria metrics the fields: `status, data, isPartial`. My guess is that the additional field `isPartial` is source of the problem.

**Comment by jacksontj** [comment]
Additional fields _shouldn't_ be an issue; would it be possible to add here the query and output against both the working and not-working endpoint (prom and VM)? My guess is this is related to the caching in VM where it returns timestamps that are off (can disable this -- https://github.com/jacksontj/promxy/blob/master/cmd/promxy/config.yaml#L55).

**Comment by oOHenry** [comment]
Query: `$prom_or_victoria/api/v1/series?match%5B%5D=prometheus_build_info&start=1615188622&end=1615210222` (found in promxy debug log after query in grafana)


Prometheus direct Query Output:
```
{
  "status": "success",
  "data": [
    {
      "__name__": "prometheus_build_info",
      "branch": "HEAD",
      "container": "prometheus",
      "endpoint": "web",
      "goversion": "go1.15.6",
      "instance": "100.64.57.230:9090",
      "job": "gocdmaster-team-helper/default-podmonitor",
      "k8s_cluster": "be-intg-iz1-bs",
      "mam_dc": "bs",
      "namespace": "gocdmaster-team-helper",
      "pod": "prometheus-team-1",
      "revision": "e4487274853c587717006eeda8804e597d120340",
      "tenant": "team",
      "version": "2.24.1"
    },
    {
      "__name__": "prometheus_build_info",
      "branch": "HEAD",
      "container": "prometheus",
      "endpoint": "web",
      "goversion": "go1.15.6",
      "instance": "100.67.127.67:9090",
      "job": "gocdmaster-team-helper/default-podmonitor",
      "k8s_cluster": "be-intg-iz2-bap",
      "mam_dc": "bap",
      "namespace": "gocdmaster-team-helper",
      "pod": "prometheus-team-1",
      "revision": "e4487274853c587717006eeda8804e597d120340",
      "tenant": "team",
      "version": "2.24.1"
    },
    {
      "__name__": "prometheus_build_info",
      "branch": "HEAD",
      "container": "prometheus",
      "endpoint": "web",
      "goversion": "go1.15.6",
      "instance": "100.67.65.145:9090",
      "job": "gocdmaster-team-helper/default-podmonitor",
      "k8s_cluster": "be-intg-iz2-bap",
      "mam_dc": "bap",
      "namespace": "gocdmaster-team-helper",
      "pod": "prometheus-team-0",
      "revision": "e4487274853c587717006eeda8804e597d120340",
      "tenant": "team",
      "version": "2.24.1"
    },
    {
      "__name__": "prometheus_build_info",
      "branch": "HEAD",
      "container": "prometheus",
      "endpoint": "web",
      "goversion": "go1.15.6",
      "instance": "100.69.147.203:9090",
      "job": "gocdmaster-team-helper/default-podmonitor",
      "k8s_cluster": "be-intg-iz1-bs",
      "mam_dc": "bs",
      "namespace": "gocdmaster-team-helper",
      "pod": "prometheus-team-0",
      "revision": "e4487274853c587717006eeda8804e597d120340",
      "tenant": "team",
      "version": "2.24.1"
    },
    {
      "__name__": "prometheus_build_info",
      "branch": "HEAD",
      "goversion": "go1.15.6",
      "instance": "prom02.srv.lan:443",
      "job": "prometheus",
      "revision": "e4487274853c587717006eeda8804e597d120340",
      "version": "2.24.1"
    },
    {
      "__name__": "prometheus_build_info",
      "branch": "HEAD",
      "goversion": "go1.15.6",
      "instance": "prom01.srv.lan:443",
      "job": "prometheus",
      "revision": "e4487274853c587717006eeda8804e597d120340",
      "version": "2.24.1"
    },
    {
      "__name__": "prometheus_build_info",
      "branch": "HEAD",
      "goversion": "go1.15.8",
      "instance": "prom02.srv.lan:443",
      "job": "prometheus",
      "revision": "a6be548dbc17780d562a39c0e4bd0bd4c00ad6e2",
      "version": "2.25.0"
    },
    {
      "__name__": "prometheus_build_info",
      "branch": "HEAD",
      "goversion": "go1.15.8",
      "instance": "prom01.srv.lan:443",
      "job": "prometheus",
      "revision": "a6be548dbc17780d562a39c0e4bd0bd4c00ad6e2",
      "version": "2.25.0"
    }
  ]
}
```
Victoria Output direct Query: 
```
{
  "status": "success",
  "isPartial": false,
  "data": [
    {
      "__name__": "prometheus_build_info",
      "namespace": "gocdmaster-team-helper",
      "job": "gocdmaster-team-helper/default-podmonitor",
      "instance": "100.67.65.145:9090",
      "pod": "prometheus-team-0",
      "tenant": "team",
      "branch": "HEAD",
      "container": "prometheus",
      "endpoint": "web",
      "goversion": "go1.15.6",
      "k8s_cluster": "be-intg-iz2-bap",
      "mam_dc": "bap",
      "revision": "e4487274853c587717006eeda8804e597d120340",
      "version": "2.24.1"
    },
    {
      "__name__": "prometheus_build_info",
      "namespace": "gocdmaster-team-helper",
      "job": "gocdmaster-team-helper/default-podmonitor",
      "instance": "100.64.57.230:9090",
      "pod": "prometheus-team-1",
      "tenant": "team",
      "branch": "HEAD",
      "container": "prometheus",
      "endpoint": "web",
      "goversion": "go1.15.6",
      "k8s_cluster": "be-intg-iz1-bs",
      "mam_dc": "bs",
      "revision": "e4487274853c587717006eeda8804e597d120340",
      "version": "2.24.1"
    },
    {
      "__name__": "prometheus_build_info",
      "namespace": "gocdmaster-team-helper",
      "job": "gocdmaster-team-helper/default-podmonitor",
      "instance": "100.67.127.67:9090",
      "pod": "prometheus-team-1",
      "tenant": "team",
      "branch": "HEAD",
      "container": "prometheus",
      "endpoint": "web",
      "goversion": "go1.15.6",
      "k8s_cluster": "be-intg-iz2-bap",
      "mam_dc": "bap",
      "revision": "e4487274853c587717006eeda8804e597d120340",
      "version": "2.24.1"
    },
    {
      "__name__": "prometheus_build_info",
      "namespace": "gocdmaster-team-helper",
      "job": "gocdmaster-team-helper/default-podmonitor",
      "instance": "100.69.147.203:9090",
      "pod": "prometheus-team-0",
      "tenant": "team",
      "branch": "HEAD",
      "container": "prometheus",
      "endpoint": "web",
      "goversion": "go1.15.6",
      "k8s_cluster": "be-intg-iz1-bs",
      "mam_dc": "bs",
      "revision": "e4487274853c587717006eeda8804e597d120340",
      "version": "2.24.1"
    },
    {
      "__name__": "prometheus_build_info",
      "job": "prometheus",
      "instance": "prom02.srv.lan:443",
      "branch": "HEAD",
      "goversion": "go1.15.8",
      "revision": "a6be548dbc17780d562a39c0e4bd0bd4c00ad6e2",
      "version": "2.25.0"
    },
    {
      "__name__": "prometheus_build_info",
      "job": "prometheus",
      "instance": "prom01.srv.lan:443",
      "branch": "HEAD",
      "goversion": "go1.15.6",
      "revision": "e4487274853c587717006eeda8804e597d120340",
      "version": "2.24.1"
    },
    {
      "__name__": "prometheus_build_info",
      "job": "prometheus",
      "instance": "prom01.srv.lan:443",
      "branch": "HEAD",
      "goversion": "go1.15.8",
      "revision": "a6be548dbc17780d562a39c0e4bd0bd4c00ad6e2",
      "version": "2.25.0"
    },
    {
      "__name__": "prometheus_build_info",
      "job": "prometheus",
      "instance": "prom02.srv.lan:443",
      "branch": "HEAD",
      "goversion": "go1.15.6",
      "revision": "e4487274853c587717006eeda8804e597d120340",
      "version": "2.24.1"
    }
  ]
}
```

**Comment by jacksontj** [comment]
I set up a local VM instance with a local prometheus instance (to scrape metrics); after doing so I ran the equivalent query (`http://localhost:8082/api/v1/series?match[]=prometheus_build_info&start=1615491658.554&end=1615495258.554`) and I was able to get results. Could you include the promxy trace level logs? Hopefully there is something in there that will help me reproduce the error.

**Comment by jacksontj** [comment]
I'm going to go ahead and close this out, if this is still an issue please feel free to re-open!

**Comment by oOHenry** [comment]
@jacksontj thanks for the hint with trace log level before I only used debug level. In trace logs I saw a that victoria metrics returned a 400 bad request. I had set `remote_read: true`,  after setting it to false it works. But I'm still confused why it worked with 0.0.62 and `remote_read: true` and breaks after I upgraded to  0.0.62.
It would be nice to log non 200 http responses as a warning instead of trace.

**Comment by jacksontj** [comment]
Thanks for the report! That actually is a larger error handling issue introduced in v0.0.63. I have pushed a fix in #417 and cut a release (https://github.com/jacksontj/promxy/releases/tag/v0.0.70) to fix this.

**Comment by jnsrikanth** [comment]
Hi,
@jacksontj 
Do we still need to set remote_read: false in order to fix this issue? Can you throw some light on it?
I am getting a Grafana Error "Template variable service failed - Bad Gateway" for some of the dashboards in Grafana.
However my Node Exporter Metrics Dashboard is working fine?
Thanks,
